src/System.Memory/src/System/SpanExtensions.cs

```diff
@@ -32,6 +31,10 @@ public static int IndexOf<T>(this Span<T> span, T value)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static int IndexOf(this Span<byte> span, byte value)
         {
+            if (!BitConverter.IsLittleEndian)
```

Comment:
👍 For keeping a big-endian path working. It would be nice if we could test on big-endian architectures.

In Kestrel, we declared big-endian bankruptcy a while ago, so Kestrel now throws on startup if it detects that it's running on a big-endian machine. I don't think that's really an option for corefx APIs.